.github/workflows/publish_charm.yaml

```diff
@@ -42,6 +42,11 @@ jobs:
   branch-up-to-date-check-enabled:
     runs-on: ubuntu-22.04
     steps:
+      - env:
+          BRANCH: ${{ github.base_ref || github.ref }}
```

Comment:
I don't think this will work if the workflow is triggered by the `release` event, as `github_ref` will be equal to a tag (which cannot have branch protection with status check). We should add a short explanation to the docs with the supported events.
https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

Reply:
Yes, but my point is that other users might try to trigger the workflow on the `release` event, which might fail. Maybe an edge case, but I still think it is worth adding a one-liner in the docs about which workflow events are supported (push, workflow_dispatch, pull_request) to avoid having to troubleshoot this in the future if someone complains.

Reply:
@addyess , if you could add a small explanation with this in the readme that'd be great